android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java

```diff
@@ -145,6 +145,14 @@ public boolean onCreateOptionsMenu(Menu menu) {
         searchView.setQuery(searchText, false);
         return true;
     }
+    @Override
+    public void onBackPressed(){
+        if((mSessions.isEmpty())&&(sessionRecyclerView.getVisibility()==View.GONE)){
+            noSessionsView.setVisibility(View.VISIBLE);
```

Comment:
@mahikaw Sorry for late reply. I tried just using `mSessions.isEmpty()`, and it worked for most use cases. Can you confirm a use case where it won't, because if it is so, this too is a bug

Because `sessionRecyclerView` should always be `View.GONE` when `mSessions.isEmpty()`